distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Dialect.enso

```diff
@@ -302,13 +309,69 @@ type SQLServer_Dialect
+       that have been used in the expression and need to be checked for nulls.
+    generate_expression self base_gen expr = case expr of
+        SQL_Expression.Column _ _ ->
+            wrapped_name = base_gen.generate_expression self expr
+            pair wrapped_name [wrapped_name]
+        SQL_Expression.Constant value -> 
+            wrapped = case value of
+                Nothing -> SQL_Builder.code "NULL"
+                _ -> SQL_Builder.interpolation value
+            pair (wrapped) [wrapped]
+        SQL_Expression.Literal value -> 
+            wrapped = SQL_Builder.code value
+            pair wrapped [wrapped]
+        SQL_Expression.Text_Literal _ ->
+            wrapped_literal = base_gen.generate_expression self expr
+            pair wrapped_literal []
+        SQL_Expression.Operation kind arguments metadata ->
+            op = self.dialect_operations.operations_dict.get kind (Error.throw <| Unsupported_Database_Operation.Error kind)
+            parsed_args_and_null_checks = arguments.map (c -> self.generate_expression base_gen c)
+            parsed_args = parsed_args_and_null_checks.map .first
+            null_checks = parsed_args_and_null_checks.map .second  . flatten
+
+            expr_result = case kind of
+                ## In the case that we actually want to check for null then we need to generate the null
+                   check sql for all the columns that have been used up to this point and that 
+                   becomes the expression.
+                "IS_NULL" -> _generate_null_check_sql_builder null_checks
+                _ -> 
+                    result = op parsed_args
+                    # If the function expects more arguments, we pass the metadata as the last argument.
+                    case result of
+                        _ : Function -> result metadata
+                        _ -> result
+            null_checks_result = if kind == "IS_NULL" then [] else null_checks
+
+            
+            pair expr_result null_checks_result
+        query : Query -> pair (base_gen.generate_sub_query self query) []
+
+    ## PRIVATE
+    generate_column self base_gen expr:(SQL_Expression | Order_Descriptor | Query) name:Text -> SQL_Builder =
+        expr_null_checks_pair = self.generate_expression base_gen expr
+        base_expr = expr_null_checks_pair.first
+        built_expr = case expr of
+            SQL_Expression.Operation _ _ _ ->
+                null_check = _generate_null_check_sql_builder expr_null_checks_pair.second
+                SQL_Builder.code "CASE WHEN " ++ null_check ++ " THEN NULL WHEN " ++ base_expr ++ " THEN 1 ELSE 0 END"
+            _ -> base_expr
+        built_expr ++ Base_Generator.alias self name
+
+## PRIVATE
+private _generate_null_check_sql_builder null_checks:Vector -> SQL_Builder =
+    (null_checks.map it->(it.paren ++ " IS NULL ")) . fold (SQL_Builder.code "1=0 ") acc-> i-> acc ++ "OR " ++ i
```

Comment:
I'd prefer to use `reduce` here instead of `fold` to avoid the `1=0` expression. Our SQL expressions are already huge and complex - I'd prefer to avoid adding additional complexity where it is not strictly necessary.

Reply:
Done